app/Events/VoucherHistoryEmailRequested.php

```diff
@@ -18,17 +18,23 @@ class VoucherHistoryEmailRequested
 
     public $user;
     public $trader;
+    public $vouchers;
+    public $date;
+    public $max_date;
     public $file;
 
     /**
      * Create a new event instance.
      *
      * @return void
      */
-    public function __construct(User $user, Trader $trader, $file)
+    public function __construct(User $user, Trader $trader, $vouchers, $file, $date, $max_date = null)
```

Comment:
Well in the "VoucherPaymentRequested" event we pass in the vouchers directly, I was following the example set by that event. I think we could probably get the vouchers from the trader. I don't think it's necessarily worth the hassle.